Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -9,44 +9,69 @@
 namespace UnityEngine.InputSystem.Editor
 {
     // Placeholder for converting InputManager.asset actions into regular asset to support conversion from 1.8.0-pre1 and 1.8.0-pre2 to asset based Project-wide actions
-    // TODO Currently not used
-    internal static class ProjectSettingsProjectWideActionsAssetConverter
-    {
-        internal const string kAssetPath = "ProjectSettings/InputManager.asset";
 
-        // TODO 1. Implement reading the kAssetPath into InputActionAsset.
-        // TODO 2. Serialize as JSON and write as an .inputactions file into Asset directory.
-        // TODO Consider preserving GUIDs to potentially enable references to stay intact.
-        // TODO 3. Let InputActionImporter do its job on importing and configuring the asset.
-        // TODO 4. Assign to InputSystem.actions
+    internal static class ProjectWideActionsAsset
+    {
+        private const string kDefaultAssetPath = "Assets/InputSystem_Actions.inputactions";
+        private const string kDefaultTemplateAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.json";
 
-        internal static void DeleteActionAssetAndActionReferences()
+        internal static class ProjectSettingsProjectWideActionsAssetConverter
         {
-            var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
-            if (objects == null)
-                return;
+            internal const string kAssetPath = "ProjectSettings/InputManager.asset";
+            internal const string kAssetName = InputSystem.kProjectWideActionsAssetName;
+
+            // DONE 1. Implement reading the kAssetPath into InputActionAsset.
+            // DONE 2. Serialize as JSON and write as an .inputactions file into Asset directory.
+            // TODO Consider preserving GUIDs to potentially enable references to stay intact.
+            // TODO 3. Let InputActionImporter do its job on importing and configuring the asset.
+            // TODO 4. Assign to InputSystem.actions
 
-            // Handle deleting all InputActionAssets as older 1.8.0 pre release could create more than one project wide input asset in the file
-            foreach (var obj in objects)
+            class ProjectSettingsPostprocessor : AssetPostprocessor
             {
-                if (obj is InputActionReference)
+                static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
```

Comment:
This will be retriggered on every asset import. I believe we could keep a project specific setting here or at least a static to avoid excessive unnecessary processing?

Reply:
@AlexTyrer

Reply:
I've added a static as a quick fix here (aee68129f9765b77ca559bcc0c9c39aa3f7b1a72)